src/App.tsx

```diff
@@ -1,24 +1,154 @@
+  const activeTodos = activeTodosCount(todos).length;
+  const completedTodos = visibeTodos.length - activeTodos;
+  const completedTodoList = filteredTodos(todos, Filter.Completed);
+
+  useEffect(() => {
+    const loadTodos = async () => {
+      try {
+        const loadedTodos = await getTodos(USER_ID);
+
+        setTodos(loadedTodos);
+      } catch (err) {
+        throw new Error();
+      } finally {
+        setIsSelected(false);
+      }
+    };
+
+    loadTodos();
+  }, []);
+
+  const setNotification = (message: Errors) => {
+    setErrorNotification(message);
+    setError(true);
+  };
+
+  const clearNotification = useCallback(() => {
+    setError(false);
+  }, []);
+
+  const clearInput = () => {
+    setInputValue('');
+  };
+
+  const handleCreateTodo = useCallback((title: string) => {
+    clearNotification();
+
+    if (!title) {
+      setNotification(Errors.Empty);
+    }
+
+    if (title.trim()) {
+      const newTodo = {
+        id: 0,
+        title,
+        userId: USER_ID,
+        completed: false,
+      };
+
+      setTempTodo(newTodo);
+      addTodo(USER_ID, newTodo)
+        .then((todo) => {
+          setTodos((prevTodos) => [...prevTodos, todo]);
+        })
+        .catch(() => {
+          setNotification(Errors.ErrorOnAdd);
+        })
+        .finally(() => {
+          clearInput();
+          setTempTodo(null);
+        });
+    }
+  }, []);
+
+  const handleDeleteTodo = useCallback((todoDel: Todo) => {
+    setDeletingId(prev => [...prev, todoDel.id]);
+
+    deleteTodo(USER_ID, todoDel.id)
+      .then(() => {
+        setTodos(prevTodos => (
+          prevTodos.filter(todo => todo.id !== todoDel.id)
+        ));
+      })
+      .catch(() => {
+        setNotification(Errors.ErrorOnDelete);
+      })
+      .finally(() => {
+        setDeletingId(prev => prev.filter(id => id !== todoDel.id));
+      });
+    clearNotification();
+  }, []);
+
+  const handleDeleteCompleted = useCallback(() => {
+    clearNotification();
+
+    completedTodoList.forEach(todo => handleDeleteTodo(todo));
+  }, [todos]);
 
   return (
-    <section className="section container">
-      <p className="title is-4">
-        Copy all you need from the prev task:
-        <br />
-        <a href="https://github.com/mate-academy/react_todo-app-loading-todos#react-todo-app-load-todos">React Todo App - Load Todos</a>
-      </p>
-
-      <p className="subtitle">Styles are already copied</p>
-    </section>
+    <div className="todoapp">
+      <h1 className="todoapp__title">todos</h1>
+
+      <div className="todoapp__content">
+        <TodoHeader
+          setInputValue={setInputValue}
+          inputValue={inputValue}
+          createTodo={handleCreateTodo}
+          tempTodo={tempTodo}
+        />
+
+        {isSelected && <Loader />}
+        {visibeTodos.length > 0 && (
+          <TodoList
+            todos={visibeTodos}
+            handleDeleteTodo={handleDeleteTodo}
+            tempTodo={tempTodo}
```

Comment:
why do we need to pass all tempTodo? just pass status to this input `disabled={!!tempTodo}`